### DISCUSSION TITLE
28BYJ-48 stepper with Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by astroisk** [aaronwmorris/indi-allsky#1712] [General]
Hi,

I was struggling to get the 28BYJ-48 stepper working the with the Raspberry Pi 5, even though the stepper works just fine with my other Python projects. For whatever reason, Raspberry Pi OS ships with `python3-rpi.gpio` which doesn't work due to different hardware setup, and should be hot-swapped with `rpi-lgpio`. The former is removed with `sudo apt remove python3-rpi.gpio` but for the latter I had to edit `requirements/requirements_gpio.txt` to remove the comment from `rpi-lgpio`. Then during the installation I selected to install additional libraries and the GPIO libraries and then everything starts working.

Hopefully this helps any others facing this issue.

Thanks!

**Comment by aaronwmorris** [comment]
I recently applied some work in #1710 that will always replace rpi.gpio with rpi.lgpio.  indi-allsky does not use the system python modules, this has to be done in the indi-allsky virtualenv.

```
source virtualenv/indi-allsky/bin/activate

pip uninstall RPi.GPIO rpi.lgpio

pip install --upgrade rpi.lgpio
```
